src/linux_tcp/tcpconnection.cpp

```diff
@@ -23,10 +23,11 @@ namespace AMQP {
  *  Constructor
  *  @param  handler         User implemented handler object
  *  @param  hostname        The address to connect to
+ *  @param  random          Randomly select one of the IP addresses that belong to the hostname
  */
-TcpConnection::TcpConnection(TcpHandler *handler, const Address &address) :
+TcpConnection::TcpConnection(TcpHandler *handler, const Address &address, bool random) :
```

Comment:
Instead of adding endless number of parameters to the TcpConnection, we support parameters inside the address. Better adjust the "address" for this.